sway-lsp/src/traverse/typed_tree.rs

```diff
@@ -972,32 +972,35 @@ impl<'a> TypedTree<'a> {
                 }
             }
             EnumScrutinee {
-                call_path,
-                decl_name,
+                enum_ref,
                 variant,
                 value,
+                instantiation_call_path,
             } => {
-                let prefixes = if let Some((last, prefixes)) = call_path.prefixes.split_last() {
-                    // the last prefix of the call path is not a module but a type
-                    if let Some(mut token) = self
-                        .ctx
-                        .tokens
-                        .try_get_mut(&to_ident_key(last))
-                        .try_unwrap()
-                    {
-                        token.typed = Some(TypedAstToken::TypedScrutinee(scrutinee.clone()));
-                        token.type_def = Some(TypeDefinition::Ident(decl_name.clone()));
-                    }
-                    prefixes
-                } else {
-                    &call_path.prefixes
-                };
+                let prefixes =
+                    if let Some((last, prefixes)) = instantiation_call_path.prefixes.split_last() {
+                        // the last prefix of the call path is not a module but a type
+                        if let Some(mut token) = self
+                            .ctx
+                            .tokens
+                            .try_get_mut(&to_ident_key(last))
+                            .try_unwrap()
+                        {
+                            token.typed = Some(TypedAstToken::TypedScrutinee(scrutinee.clone()));
+                            token.type_def = Some(TypeDefinition::Ident(
+                                enum_ref.name().clone(), // prefixes.last().cloned().unwrap_or(enum_ref.name().clone()),
```

Comment:
Is the comment required?

Reply:
Oops, this is a mistake. Will remove, thanks for catching this.